src/component/handlers/edit/editOnBeforeInput.ts

```diff
@@ -60,9 +61,20 @@ function replaceText(
   entityKey: string | null,
   forceSelection: boolean,
 ): EditorState {
+  // Special case '. ': Replace ' ' with '. '
+  // https://github.com/facebook/draft-js/issues/2422#issuecomment-683221263
+  let selection = editorState.selection;
+
+  if (text === '. ') {
+    selection = makeSelectionState({
+      ...selection,
+      anchorOffset: selection.anchorOffset - 1,
+    });
+  }
```

Comment:
It's probably worth checking here that the new selection we're creating is valid and meets these specific error conditions. Maybe:

```suggestion
  if (text === '. ' && selection.anchorKey === selection.focusKey && selection.anchorOffset > 0 && selection.anchorOffset === selection.focusOffset + 1) {
    selection = makeSelectionState({
      ...selection,
      anchorOffset: selection.anchorOffset - 1,
    });
  }
```

(Is that what the selection will look like in this case?)

Reply:
in this case anchorOffset and focusOffset are the same